server/service/endpoint_utils.go

```diff
@@ -23,6 +24,122 @@ func makeDecoderForType(v interface{}) func(ctx context.Context, r *http.Request
 	}
 }
 
+func makeDecoderForIDs(v interface{}, idKeys ...string) func(ctx context.Context, r *http.Request) (interface{}, error) {
```

Comment:
Now... this is, to me, the best part of this PR.

One thing that I like about the new approach is that I didn't have to write a decoder for a while for any of the new endpoints we added. These APIs needed smarter decodes, so here we go.

The idea is that we now have our own struct tag (yay!): `url`, for things that come as part of the URL path or query args.

Reply:
Would love to see some code comments here as it's quite hard for me to follow use of `reflect`. Some unit tests to further specify/test the behavior would also be nice. I'm fine with following up in a separate PR for those things so we can get this working API to @RachelElysia.